src/parser/pegen.ts

```diff
@@ -1,27 +1,2188 @@
+//         }
+//         return NULL;
+//     }
+//     // This transfers the ownership to the tokenizer
+//     tok->filename = filename_ob;
+//     Py_INCREF(filename_ob);
+
+//     // We need to clear up from here on
+//     mod_ty result = NULL;
+
+//     int parser_flags = compute_parser_flags(flags);
+//     int feature_version = flags && (flags->cf_flags & PyCF_ONLY_AST) ?
+//         flags->cf_feature_version : PY_MINOR_VERSION;
+//     Parser *p = _PyPegen_Parser_New(tok, start_rule, parser_flags, feature_version,
+//                                     NULL, arena);
+//     if (p == NULL) {
+//         goto error;
+//     }
+
+//     result = _PyPegen_run_parser(p);
+//     _PyPegen_Parser_Free(p);
+
+// error:
+//     PyTokenizer_Free(tok);
+//     return result;
+// }
+
+// void *
+// _PyPegen_interactive_exit(Parser *p)
+// {
+//     if (p->errcode) {
+//         *(p->errcode) = E_EOF;
+//     }
+//     return NULL;
+// }
+
+/* Creates a single-element asdl_seq* that contains a */
+export function singleton_seq(a: AST): AST[] {
+    return [a];
+}
+// asdl_seq *
+// _PyPegen_singleton_seq(Parser *p, void *a)
+// {
+//     assert(a != NULL);
+//     asdl_seq *seq = _Py_asdl_seq_new(1, p->arena);
+//     if (!seq) {
+//         return NULL;
+//     }
+//     asdl_seq_SET(seq, 0, a);
+//     return seq;
+// }
+
+// /* Creates a copy of seq and prepends a to it */
+// asdl_seq *
+// _PyPegen_seq_insert_in_front(Parser *p, void *a, asdl_seq *seq)
+// {
+//     assert(a != NULL);
+//     if (!seq) {
+//         return _PyPegen_singleton_seq(p, a);
+//     }
+
+//     asdl_seq *new_seq = _Py_asdl_seq_new(asdl_seq_LEN(seq) + 1, p->arena);
+//     if (!new_seq) {
+//         return NULL;
+//     }
+
+//     asdl_seq_SET(new_seq, 0, a);
+//     for (Py_ssize_t i = 1, l = asdl_seq_LEN(new_seq); i < l; i++) {
+//         asdl_seq_SET(new_seq, i, asdl_seq_GET(seq, i - 1));
+//     }
+//     return new_seq;
+// }
+
+// /* Creates a copy of seq and appends a to it */
+// asdl_seq *
+// _PyPegen_seq_append_to_end(Parser *p, asdl_seq *seq, void *a)
+// {
+//     assert(a != NULL);
+//     if (!seq) {
+//         return _PyPegen_singleton_seq(p, a);
+//     }
+
+//     asdl_seq *new_seq = _Py_asdl_seq_new(asdl_seq_LEN(seq) + 1, p->arena);
+//     if (!new_seq) {
+//         return NULL;
+//     }
+
+//     for (Py_ssize_t i = 0, l = asdl_seq_LEN(new_seq); i + 1 < l; i++) {
+//         asdl_seq_SET(new_seq, i, asdl_seq_GET(seq, i));
+//     }
+//     asdl_seq_SET(new_seq, asdl_seq_LEN(new_seq) - 1, a);
+//     return new_seq;
+// }
+
+// static Py_ssize_t
+// _get_flattened_seq_size(asdl_seq *seqs)
+// {
+//     Py_ssize_t size = 0;
+//     for (Py_ssize_t i = 0, l = asdl_seq_LEN(seqs); i < l; i++) {
+//         asdl_seq *inner_seq = asdl_seq_GET(seqs, i);
+//         size += asdl_seq_LEN(inner_seq);
+//     }
+//     return size;
+// }
+
+/* Flattens an asdl_seq* of asdl_seq*s */
+export function seq_flatten(seqs: AST[][]): AST[] {
```

Comment:
how closely do you want to follow cpython  in this port? 

In skulpt we've generally shy-ed away from writing these helper functions. 

But I can also see it might be handy to have them around.

Reply:
Good question. I guess it's either change the grammarfile or write the helper function since these little helper functions are nested in the grammar as the action. 

At the moment I'm thinking the to include a helper functions if it's  a grammar action.

Reply:
Ah I didn't know this was called from the generated parser.